### DISCUSSION TITLE
Codemagic API - GET /builds -> workflowId always null

### DISCUSSION THREAD

**Discussion by gerken-tss** [codemagic-ci-cd/codemagic-docs#1941] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I have several workflows in my company. Every workflow represents an individualized variant of the app. Sometimes, I want to start the workflow for let's say only 6 variants. I don't want to start the same process 6 times via the Codemagic UI. Also I might want to set individual release notes for every build.

Because this is not possible via Codemagic UI, I wanted to start developing my own custom client that triggers the builds via API.

For this, I need a list of available workflows. 

According to the [docs](https://docs.codemagic.io/rest-api/builds/), the list of workflows is part of the `/builds` endpoint.

However, whenever I send the specified request to the API, my response is missing the field `workflowId` in every build. Instead, only the `fileWorkflowId` is set. However, I need the correct workflow id to match it with the workflows from the [apps](https://docs.codemagic.io/rest-api/applications/) endpoint in order to provide a list in the UI.

**Reproduction**:

Like it's described in the [docs](https://docs.codemagic.io/rest-api/builds/), send a request to https://api.codemagic.io/builds with your personal API token in the header:

```sh
  curl -H "Content-Type: application/json" \
       -H "x-auth-token: <API Token>" \
       --request GET "https://api.codemagic.io/builds"
```

### Expected results

The response contains a body like this:

```javascript
{
 …
  "workflowId": "something-useful,
  …
}
```

### Actual results

The response contains a body like this:

```javascript
{
  "_id": "…",
  "index": 8,
  "appId": "…",
  "status": "canceled",
  "version": null,
  "branch": "master",
  "startedAt": null,
  "startedBy": "Gitlab webhook",
  "createdAt": "2023-07-05T09:03:31.657000+00:00",
  "finishedAt": "2023-07-05T09:03:42.093000+00:00",
  "workflowId": null,
  "fileWorkflowId": "this-is-set",
  "instanceType": "mac_mini_m1",
  "commit": {
    "authorAvatarUrl": "…",
    "authorEmail": "…",
    "authorName": "…",
    "branch": null,
    "tag": "2.2.0+111",
    "commitMessage": "…",
    "hash": "…",
    "url": "…"
  },
  "config": {
    "name": "…",
    "buildSettings": {
      "flutterVersion": "stable",
      "platforms": [],
      "url": null,
      "xcodeVersion": "14.3.1",
      "workingDirectory": null
    }
  },
  "dynamicConfig": {},
  "artefacts": [],
  "buildActions": [
    …
  ],
  "message": null,
  "feedback": null,
  "screenshots": null,
  "pullRequest": null,
  "tag": "2.2.0+111",
  "sshAccessEnabled": false,
  "sshAccess": null,
  "vncAccess": null,
  "scheduledBuildId": null,
  "appStoreConnectTasks": [],
  "labels": []
}
```

### Build id (optional)

64a531e34cff37cffb483b7a

**Comment by kulakarla** [comment, marked as answer]
Hello @gerken-tss 

The reason you are not able to retrieve the workflowId-s are the workflow names declared inside your YAML, which the API does not read. It is possible to get the ID-s when using WFE, but as for YAML, you have to find an alternative way. Maybe read your YAML locally and get the workflow id-s from there. The workflow id-s are the workflow names, e.g `ios-workflow` and `test-workflow2` in this example:
```
workflows:
    ios-workflow:
      ... ...
    test-workflow2:
      ... ...

```

**Reply by gerken-tss** [reply]
@kulakarla 
That makes total sense as the workflow-ids are never introduced to the web (at least not as a whole. Only partly when executing a particular build). Thank you, I will think of a solution! 🙂
